core/identity/identity.go

```diff
@@ -280,6 +278,32 @@ func (s *service) getIdentitiesDataFromRepo(ctx context.Context, identities []st
 	return res, nil
 }
 
+func (s *service) batchIdentities(ctx context.Context, identities []string) ([]*identityrepoproto.DataWithIdentity, []string) {
+	resultIdentities := make([]*identityrepoproto.DataWithIdentity, 0, len(identities))
```

Comment:
Just summing all the comments, there should be one abstract function, which splits the []T to several batches of size <= N (take N as 200 pls), this function should be in package `slice` or whatever and tested

and then call in parallel some named func for every batch which returns results using waitgroup, gather everything and return.

This func should either get stuff from remote or if nothing in remote get stuff from local.

I would do it like this